### PR TITLE
Support fetching a single page

### PR DESCRIPTION
**Pull request by mquan**
This change allows user to fetch data for a specific page by providing `params[:page]`

### CODE REVIEW

File: lib/greenhouse_api/client.rb [hunk 1]
```diff
@@ -59,13 +59,13 @@ def list_many(resource, params = {})
           http_method: :get,
           headers: headers,
           endpoint: resource,
-          params: params.merge(page: page, per_page: per_page)
+          params: params.merge(page: page, per_page: params[:per_page] || per_page)
```
Comment (tannerljohnson) [inline lib/greenhouse_api/client.rb:62]: should we compute `per_page` on line 56 where we're already applying some logic there?

**Review by tannerljohnson — APPROVED** [review]
nice! just a quick question for cleaning up the per_page method
i don't think it's blocking though

File: lib/greenhouse_api/client.rb (hunk 1 above)
Comment (mquan) [reply lib/greenhouse_api/client.rb:62]: great callout, we want to use `per_page` over `limit` and I originally had it like this

```ruby
per_page = params[:per_page] || (limit ? [limit - data.length, MAX_PER_PAGE].min : MAX_PER_PAGE)
``` 

But I think it's important to keep the computation in one place so I'll try a multiline approach.

Comment (tannerljohnson) [reply lib/greenhouse_api/client.rb]: nice - new code looks clean

**Pull request merged**